test/applyMiddlewareTests.ts

```diff
@@ -1,65 +1,64 @@
 import 'jasmine';
-import applyMiddleware from '../src/applyMiddleware';
-import * as dispatcher from '../src/dispatcher';
-import { getGlobalContext, __resetGlobalContext } from '../src/globalContext';
+import { createSatchel, createSatchelInternal, SatchelInstance } from '../src/createSatchel';
+import { createTestSatchel } from './utils/createTestSatchel';
 
 describe('applyMiddleware', () => {
     it('updates dispatchWithMiddleware to point to the middleware pipeline', () => {
         // Arrange
-        __resetGlobalContext();
         let testMiddleware = jasmine.createSpy('testMiddleware');
+        const satchel = createTestSatchel({ middleware: [testMiddleware] });
 
         // Act
-        applyMiddleware(testMiddleware);
-        getGlobalContext().dispatchWithMiddleware({});
+        satchel.__dispatchWithMiddleware({});
 
         // Assert
         expect(testMiddleware).toHaveBeenCalled();
     });
 
     it('the action message and next delegate get passed to middleware', () => {
         // Arrange
-        __resetGlobalContext();
-
         let dispatchedActionMessage = {};
         let actualNext;
         let actualActionMessage;
 
-        applyMiddleware((next: any, actionMessage: any) => {
+        const testMiddleware = (next: any, actionMessage: any) => {
             actualNext = next;
             actualActionMessage = actionMessage;
-        });
+        };
+        const satchel = createTestSatchel({ middleware: [testMiddleware] });
 
         // Act
-        getGlobalContext().dispatchWithMiddleware(dispatchedActionMessage);
+        satchel.__dispatchWithMiddleware(dispatchedActionMessage);
 
         // Assert
         expect(actualActionMessage).toBe(dispatchedActionMessage);
-        expect(actualNext).toBe(dispatcher.finalDispatch);
+        expect(actualNext).toBe(satchel.__finalDispatch);
     });
 
     it('middleware and finalDispatch get called in order', () => {
         // Arrange
-        __resetGlobalContext();
         let sequence: string[] = [];
 
-        spyOn(dispatcher, 'finalDispatch').and.callFake(() => {
-            sequence.push('finalDispatch');
-        });
-
-        applyMiddleware(
+        const middleware = [
             (next: any, actionMessage: any) => {
                 sequence.push('middleware1');
                 next(actionMessage);
             },
             (next: any, actionMessage: any) => {
                 sequence.push('middleware2');
                 next(actionMessage);
-            }
+            },
+        ];
+
+        const satchel = createTestSatchel(
+            { middleware },
+            jasmine.createSpy('finalDispatch').and.callFake(() => {
```

Comment:
Can't you directly spy on `satchel.__finalDispatch` so that you don't need this extra test-only argument?

Reply:
So it couldn't spy on __finalDispatch since it's only used to generate `dispatchWithMiddleware`.  I altered the test to test the generation of `dispatchWithMiddleware` instead to test the behavior of the original test.